stan/math/rev/core/vari.hpp

```diff
@@ -53,13 +63,35 @@ class vari {
    * derivative propagation, the chain() method of each variable
    * will be called in the reverse order of construction.
    *
+   * @tparam S an Arithmetic type.
    * @param x Value of the constructed variable.
    */
-  explicit vari(double x) : val_(x), adj_(0.0) {
+  template <typename S,
+            std::enable_if_t<std::is_convertible<S&, Scalar>::value>* = nullptr>
```

Comment:
Do you mean the template `S` here? Or in general handling `vari_value<float/double/long double>`?

Reply:
I think the template `S`.

I guess I'm happy with `vari_value<float>` `vari_value<long double>` existing, just a plan for doing anything beyond that requires matrix discussions (especially if we're going to make an autodiff land where all these things live together).

Reply:
Skip the "I guess" part. I am happy with them, I'm just really unsure about automatic conversions for everything.

Reply:
>  I'm just really unsure about automatic conversions for everything.

Yeah I get that, right now tho' I'm just not sure if we are going to want those conversions. When would we use them? If we keep them can bad things happen? idt anything bad happens, and I'm unsure of if we will need them in the future, so I sort of figure why not ya know?

Reply:
> so I sort of figure why not ya know?

Well I was scared that this would influence the matrix implementation, but the matrix `vari_value` s are a totally different class so that doesn't actually matter so much.

How about this for an argument: there's no reason for them to be there and so we shouldn't have them. We should just stick with `T` type args (so we'll get the regular C++ 1 cast) until we're ready to invest in doing mixed precision autodiff.

Reply:
Yeah that's fair aight I'll rip these out

Reply:
Done!